structured.js

```diff
@@ -505,12 +506,53 @@
             }
             // Recursively check for matches
             if ((_.isArray(currTree[key]) &&
-                    checkNodeArray(currTree[key], toFind, peersToFind, wVars, matchResults, options)) ||
+                    checkNodeArray(currTree[key], toFind, peersToFind, wVars, matchResults, options, false)) ||
                 (!_.isArray(currTree[key]) &&
-                    checkMatchTree(currTree[key], toFind, peersToFind, wVars, matchResults, options))) {
+                    checkMatchTree(currTree[key], toFind, peersToFind, wVars, matchResults, options, false))) {
                 return matchResults;
             }
         }
+        if (!recursing) {
+            var mod = restructureTree(currTree, toFind, peersToFind, wVars, matchResults, options);
+            if (mod) {
+                return checkMatchTree(mod, toFind, peersToFind, wVars, matchResults, options, true);
+            }
+        }
+        return false;
+    }
+    
+    /*
+        Applies simple transformations to the parse tree to reattempt matching
+        Takes an argument list identical to checkMatchTree() above
```

Comment:
Can you add a simple example to this comment as well, like you had in the PR, like "this expression is turned into this expression"?